src/components/Dropdown/Dropdown.test.js

```
@@ -0,0 +1,116 @@
+import React from 'react'
+import { mount, shallow } from 'enzyme'
+import Dropdown from './Dropdown'
+import DropdownTrigger from './DropdownTrigger'
+import DropdownMenu from './DropdownMenu'
+import DropdownItem from './DropdownItem'
+import DropdownDivider from './DropdownDivider'
+
+describe('Dropdown', () => {
+    it('renders Dropdown Component without props', () => {
+        const component = shallow(
+            <Dropdown></Dropdown>
+        ) 
+        expect(component).toMatchSnapshot()
+        expect(component).toHaveLength(1)
+    });
+
+    it('renders Dropdown Component with valid props', () => {
+        const component = shallow(
+            <Dropdown align="right" initialActive={true} onChange={()=>console.log("testing")}></Dropdown>
+        )
+        expect(component).toHaveLength(1)
+        expect(component.find('.is-right')).toHaveLength(1)
+        expect(component.find('.is-active')).toHaveLength(1)
+    });
+    
+    it('renders DropdownItem Component with invalid props', () => {
+        spyOn(console, 'error') //suppress error in console output
+        const component = shallow(
+            <Dropdown align={false} initialActive="bad value" onChange="bad value"></Dropdown>
```

Should the dropdown render with the wrong values?  If not, then what you want to test is that when this component renders it throws an error.  You can use jest to check of an error was thrown.

---

I was looking for this functionality but was unable to find it yet. let me know if you have an idea of how to confirm the results of console.error

---

What kind of error throwing should occur in a React component, aside from the console.error messages generated by prop-types?

---

If a component absolutely can't function because of some internal state or prop mismatch, then it should throw an error. ie., `throw new Error('Description of the error here')`.  By throwing errors when something unrecoverable happens, that allows us to write a catch handler at the top level of the application to handle notifying users about catastrophic or unrecoverable failures.

---

The component  and sub components currently will be able to function as a series of divs regardless of what is passed in, so nothing would be game breaking. The only exception is that DropdownToggle and DropdownItem absolutely require to be wrapped with Dropdown for context